specs/Menu.md

```diff
@@ -1,731 +1,1068 @@
-   </div>
-   <div tabindex="-1" class="contextMenuItem-3-1-50" role="menuitem">
-       <span class="contextMenuItem_contentRegion-3-1-51">Item three</span>
-   </div>
-   <div tabindex="-1" class="contextMenuItem-3-1-50" role="menuitem">
-       <span class="contextMenuItem_contentRegion-3-1-51">SubItem one</span>
-   </div>
-   <div tabindex="-1" class="contextMenuItem-3-1-50" role="menuitem">
-       <span class="contextMenuItem_contentRegion-3-1-51">SubItem Two</span>
-   </div>
-</div>
+`Popup` provider 3 different properties to handle bounds and overflow:
+
+- flipBoundary - the bounds to calculate when to flip positioning of the popup
+- overflowBoundary - the bounds to shift the popup without overflowing
+- mountNode - where the popup is actually rendered in the DOM, by default this is a portal to a div in body
+
+```typescript
+<ContextualMenu
+  // pixel values for bounding rectangle
+  // defaults to target window as default bounding rectangle
+  bounds={{height: 0, width: 0, top: 0, left:0 , right: 0, bottom: 0}}
+  // callback for bounds
+  bounds{(target, targetWindow) => ({/*Same object as above*/})}
+  target={htmlElement}
+
+  // renders to a portal node on body
+  layerProps={/*ILayerProps*/}
+
+  // every single one of the above can all be declared here too
+  calloutProps={{bounds, target}}
+/>
 ```
 
-#### Comments
+`ContextualMenu` or `Callout` has no notion of separate boundaries for flip or overflow, and auto behaviour is used for flip and overflow 'pushing'
+It should also be noted that `ContextualMenu` allows all of the same props to be passed to its submenus to custom tweak position for each submenu if necessary
 
-Fast-DNA is one of the only menus that doesn't use `li` elements.
+#### Submenu positioning
 
-### MaterialUI
+The default positioning for both v0 and v7 submenus is:
 
-_Note:_ Some long class names removed
+- rightTopEdge (v7)
+- top-after (v0)
 
-```HTML
-<ul class="MuiList-root MuiList-padding" role="menu" tabindex="-1" id="simple-menu">
-   <li tabindex="0" role="menuitem" aria-disabled="false">Item one<span class="MuiTouchRipple-root"></span></li>
-   <hr class="MuiDivider-root">
-   <li tabindex="-1" role="menuitem" aria-disabled="false">Item two<span class="MuiTouchRipple-root"></span></li>
-   <li tabindex="-1" role="menuitem" aria-disabled="false">Item three<span class="MuiTouchRipple-root"></span></li>
-   <li tabindex="-1" role="menuitem" aria-disabled="false">Subitem one<span class="MuiTouchRipple-root"></span></li>
-   <li tabindex="-1" role="menuitem" aria-disabled="false">Subitem Two<span class="MuiTouchRipple-root"></span></li>
-</ul>
+Both will also flip appropriately when the overflow boundary is too small.
+
+The main difference between the two is that v0 submenu's position does not expose any way to customize or override the positioning of the submenu. However v7 allows every single customization as the root menu. It is very possible to do the below:
```

Comment:
I **think** it is possible to pass popper props to submenu. @layershifter is it correct?

Reply:
Looking at our code I don't think so:

https://github.com/microsoft/fluentui/blob/89eb6bc42cdc8d49277007e2268cd52f0beced6c/packages/fluentui/react-northstar/src/components/Menu/MenuItem.tsx#L504

Reply:
It is possible but is actually not present in our documentation, it's a hidden prop you can add `menu.popper` https://fluentsite.z22.web.core.windows.net/0.52.0/components/menu/props

Reply:
Just FYI, hidden props mean we failed. Let's ensure we don't do this moving forward.